### PR TITLE
Do not use normal mode to replace / insert text

### PR DESCRIPTION
**Pull request by harshad1**
Using normal mode commands to insert characters is very slow on my laptop (I am running WSL on windows).

In this PR, I have replaced normal-mode functionality with vim functions, which are faster.

This PR is still a work in progress. I am creating it as a place-holder and so that I can get feedback.

### CODE REVIEW

File: plugin/bullets.vim [hunk 1]
```diff
@@ -558,10 +558,8 @@ endfun
 fun! s:set_checkbox(lnum, marker)
   let l:initpos = getpos('.')
   let l:pos = s:find_checkbox_position(a:lnum)
-  let l:checkbox_content = getline(a:lnum)[l:pos]
```
Comment (harshad1) [inline plugin/bullets.vim:561]: checkbox_content appears to be unused, I removed it

File: plugin/bullets.vim [hunk 2]
```diff
@@ -1244,6 +1242,16 @@ fun! s:sibling_checkbox_status(lnum)
   return l:checkbox_markers[l:completion]
 endfun
 
+fun! s:replace_char(lnum, col, item)
+    call s:replace(a:lnum, a:col - 1, a:col + 1, a:item)
+endfun
+
+fun! s:replace(lnum, col_start, col_end, item)
```
Comment (harshad1) [inline plugin/bullets.vim:1249]: I am open to suggestions for better function names.

Comment (dkarter) [reply plugin/bullets.vim:1249]: `replace_in_line` 🤷 ?

File: plugin/bullets.vim (hunk 1 above)
Comment (dkarter) [reply plugin/bullets.vim:561]: Good catch! I'm surprised the linter (Vint) didn't catch that. It runs on builds.

**Comment by harshad1** [comment]
@dkarter this is good to go subject to further comments/feedback.

I have to say, you have the best test suite / CI system I have seen for a project of this type.

**Pull request merged**